subsys/bluetooth/host/smp.c

```diff
@@ -2440,7 +2442,8 @@ static u8_t smp_pairing_req(struct bt_smp *smp, struct net_buf *buf)
 	smp->prsp[0] = BT_SMP_CMD_PAIRING_RSP;
 	rsp = (struct bt_smp_pairing *)&smp->prsp[1];
 
-	rsp->auth_req = get_auth(req->auth_req);
+	rsp->auth_req = get_auth(req->auth_req,
+				 req->auth_req & BT_SMP_AUTH_MITM);
```

Comment:
Should we just give false here? It seems strange to me to let the request MITM flag control our response MITM flag. Unless we have the IO capabilities we cannot achieve MITM anyway. And the option to force MITM flag on should be enough to solve the problem.

Reply:
You have right regarding mitm flag. But apart from BT_SMP_ENFORCE_MITM, conn->required_sec_level has to be checked when BT_SMP_ENFORCE_MITM is disabled.

Reply:
Yes, we do that in `sec_level_reachable`, which is done before. So it will always be true, otherwise -EINVAL is returned.

Reply:
But the aim of this PR is to have a possibility to set the MITM flag based security level provided by user. Looking into API:

/** Security level. */
typedef enum __packed {
	/** Only for BR/EDR special cases, like SDP */
	BT_SECURITY_NONE,
	/** No encryption and no authentication. */
	BT_SECURITY_LOW,
	/** Encryption and no authentication (no MITM). */
	BT_SECURITY_MEDIUM,
	/** Encryption and authentication (MITM). */
	BT_SECURITY_HIGH,
	/** Authenticated Secure Connections */
	BT_SECURITY_FIPS,
} bt_security_t;

Passing BT_SECURITY_MEDIUM to bt_conn_security will set the MITM flag based on local IO Capabilities even if user requested encryption without authentication.

Reply:
Thank you for explaining, that makes sense.